osu.Game/Overlays/Notifications/ProgressNotification.cs

```diff
@@ -43,17 +68,18 @@ protected override void LoadComplete()
             updateState();
         }
 
-        private readonly CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
-
-        public CancellationToken CancellationToken => cancellationTokenSource.Token;
-
         public ProgressNotificationState State
         {
             get => state;
             set
             {
                 if (state == value) return;
 
+                // Enable back user cancelling if notification transitioned to
+                // cancelled state by the component using it. likely due to task failure.
+                if (value == ProgressNotificationState.Cancelled)
+                    Cancellable = true;
```

Comment:
I find this logic strange and I'd go as far as to say even a little wrong semantically. When you read this it makes no sense why you would allow cancelling an already cancelled notification. If anything I'd say transitioning to `Cancelled` state and setting `Cancellable = true` should *both* enable the button, but *independently* of each other.